pandas/core/base.py

```diff
@@ -819,8 +833,10 @@ def tolist(self):
 
         if is_datetimelike(self):
```

Comment:
this should be overriden in EA, rather than specific dispatching via if/else here, IOW it should be a part of the interface, or be defined as ``list(.values)``

Reply:
I am not sure it is needed to add a `tolist` to the interface, I think in general it can rely on `__iter__`/`__getitem__` (so `list(self._values)`). 
The only problem in that case is that we still need to distinguish between normal numeric types (where the above would return numpy scalars and not python scalars) and other types where this already gives the correct result.